core/os/memory.h

```diff
@@ -41,24 +41,30 @@
 #define PAD_ALIGN 16 //must always be greater than this at much
 #endif
 
+#if __SIZEOF_POINTER__ == 8 || defined(_WIN64)
+#define SIZE_TYPE uint64_t
```

Comment:
just use uint64_t, Even in 32 bits systems size_t is 64 bits, as it's used for filesystem APIs too